### DISCUSSION TITLE
Install instructions for R users

### DISCUSSION THREAD

**Discussion by albersonmiranda** [extendr/extendr#826] [Q&A]
The casual R user won't know Cargo. Once in CRAN, `install.packages()` will download precompiled binaries for Windows users, right? What about Linux and mac users?

So my question is: What should be default install instructions for Rust based R packages? And how can I include informative messages during install? Right now it just fails with "cargo not found", but being honest, average user won't read install log.

**Comment by CGMossa** [comment]
I must say this subject is often on my mind.
Pre-compiled binaries is a hot topic in both R and Rust community.

I personally don't know how to configure things to achieve it.

R-universe attempts at solving this, see [slide on r-universe](https://jeroen.github.io/runiverse2021/#55),
and plus they have several blogs on the matter https://ropensci.org/blog/2024/01/14/runiverse-arm64/,
https://ropensci.org/blog/2023/11/17/runiverse-wasm/. 

In the rust world, there is [`cargo-dist`](https://github.com/axodotdev/cargo-dist).

I personally haven't dug into these solutions. I'd be happy to assist if something comes up. We need a few more people that have gone through this route to see if extendr-ecosystem has to provide assistance there.

Also, I'd like to thank you for using the Discussion forum.

**Reply by JosiahParry** [reply]
I've come to realize that the r-universe solution isn't as _nice_ as I would hope.

See https://github.com/JosiahParry/valve/issues/14 and https://github.com/r-universe-org/help/issues/481

`ubuntu:latest` is _really_ new and probably incompatible for many people. 

We should consider using releases and a special GHA CI as a way to store precompiled binaries for R versions and OS.
